doc: update plugin-extensibility.md spec (#191)
notaryproject/notaryproject: specs/plugin-extensibility.md

**Open Item** : [Plugin install paths](https://github.com/notaryproject/notation/issues/167)

To enumerate all available plugins the `PLUGIN_DIRECTORY` is scanned based on per OS:
| OS      | PLUGIN_DIRECTORY                                     |
| ------- | ---------------------------------------------------- |
| Unix    | `$XDG_CONFIG_HOME/notation/plugins`                  |
| Windows | `%AppData%/notation/plugins`                         |
| Darwin  | `$HOME/Library/Application Support/notation/plugins` |

> Each plugin executable and dependencies are installed under directory `$PLUGIN_DIRECTORY/{plugin-name}` with an executable under that directory `$PLUGIN_DIRECTORY/{plugin-name}/notation-{plugin-name}`.

Reviewer: we dont care where dependencies are installed.
```suggestion
Each plugin executable must be located under `$PLUGIN_DIRECTORY/{plugin-name}` directory, with executable named as  `notation-{plugin-name}`.
```

Author: Updated.
